### PR TITLE
Ideas page for 2019.

### PR DESCRIPTION
**Pull request by chrizandr**
Added ideas for GSoC 2019. 

Please suggest other ideas and I will add them to the page.

### CODE REVIEW

**Comment by Mec-iS** [comment]
Maybe for every ideas we can link the issues that are already in the different repositories?

**Comment by xadahiya** [comment]
Nice idea

**Comment by chrizandr** [comment]
I'll link the issues. We just have three mentors for now right? Could you guys message me what you would be interested in working on?

File: pages/GSoC-19.md [hunk 1]
```diff
@@ -5,4 +5,79 @@ permalink: /gsoc-19
 ---
 
 
-# Ideas will go here
\ No newline at end of file
+# Ideas for GSoC 2019
+----
+
+## Work on the Hydra Client
```
Comment (Mec-iS) [inline pages/GSoC-19.md]: this is Hydra Agent

File: pages/GSoC-19.md [hunk 2]
```diff
@@ -5,4 +5,79 @@ permalink: /gsoc-19
...
+### Idea:
+Currently the client is a proof of concept and only implements GET requests. We need a full fledged client that is in compliance with the spec. We also need to make sure that the client syncs with the server to keep it's data updated. This is not currently handled by the client. The client must also contain a battery of tests to ensure that a server is in compliance with the spec or not.
+### Skill Required:
+- Python
+- Redis
+- Hydra
+- OpenCypher
+- Hydrus
+### Difficulty Level : Intermediate
```
Comment (Mec-iS) [inline pages/GSoC-19.md]: Add here the issues already in place in `python-hydra-agent`

File: pages/GSoC-19.md [hunk 3]
```diff
@@ -5,4 +5,79 @@ permalink: /gsoc-19
...
+- Creation of dynamic endpoints. Users need to be able to define endpoints that have some functionality. Right now hydrus can only serve static data and users are unable to define the way in which incoming data is processed in a hydrus server.
+- General improvements in code and optimising the server functionality.
+### Skills Required:
+- Python
+- Hydra
+- Hydrus
+- Semantic Web
+- Ability to write test suites
+- Software Engineering
+### Difficulty Level: Hard
```
Comment (Mec-iS) [inline pages/GSoC-19.md]: Add here the issues that are already in `hydrus` labeled as "gsoc 2019"

File: pages/GSoC-19.md [hunk 4]
```diff
@@ -5,4 +5,79 @@ permalink: /gsoc-19
...
 
 
-# Ideas will go here
\ No newline at end of file
+# Ideas for GSoC 2019
+----
+
+## Work on the Hydra Client
+### Idea:
+Currently the client is a proof of concept and only implements GET requests. We need a full fledged client that is in compliance with the spec. We also need to make sure that the client syncs with the server to keep it's data updated. This is not currently handled by the client. The client must also contain a battery of tests to ensure that a server is in compliance with the spec or not.
```
Comment (PrajwalM2212) [inline pages/GSoC-19.md]: `We also need to make sure that the client syncs with the server to keep it's data updated.`. 
The issue is in the hydrus repo https://github.com/HTTP-APIs/hydrus/issues/300.
I understand that the client needs to sync with the server but this feature has to implemented in the hydrus server first.

**Comment by Mec-iS** [comment]
moved to #12

**Comment by Mec-iS** [comment]
These changes have been applied

**Pull request closed without merging**